--- ceres-solver-1.8.0/CMakeLists.txt
@@ -385,10 +385,6 @@
         # At least on Linux, we need pthreads to be enabled for mutex to
         # compile.  This may not work on Windows or Android.
         FIND_PACKAGE(Threads REQUIRED)
-        SET(STATIC_LIBRARY_FLAGS
-          "${STATIC_LIBRARY_FLAGS} ${CMAKE_THREAD_LIBS_INIT}")
-        SET(CMAKE_SHARED_LINKER_FLAGS
-          "${CMAKE_SHARED_LINKER_FLAGS} ${CMAKE_THREAD_LIBS_INIT}")
         ADD_DEFINITIONS(-DCERES_HAVE_PTHREAD)
         ADD_DEFINITIONS(-DCERES_HAVE_RWLOCK)
       ENDIF (UNIX)
+++ ceres-solver-1.8.0/internal/ceres/CMakeLists.txt
@@ -161,6 +161,7 @@
 IF (OPENMP_FOUND)
   IF (NOT MSVC)
     LIST(APPEND CERES_LIBRARY_DEPENDENCIES gomp)
+    LIST(APPEND CERES_LIBRARY_DEPENDENCIES ${CMAKE_THREAD_LIBS_INIT})
   ENDIF (NOT MSVC)
 ENDIF (OPENMP_FOUND)
